### PR TITLE
Refine validate_merge_block unit tests

### PR DESCRIPTION
**Pull request by mkalinin**
### Change set
* Breaks `test_terminal_validity` into two groups of tests:
    * `test_is_valid_terminal_pow_block`
    * `test_validate_merge_block`
* Adjusts tests to the new signature and semantics of `get_pow_block` function
* Adds terminal block hash checks
* A bit of refactor
    * Moves `PowBlock` related helpers to the new `pow_block.py`

### CODE REVIEW

File: tests/core/pyspec/eth2spec/test/helpers/pow_block.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+from random import Random
+from eth2spec.utils.ssz.ssz_typing import uint256
+
+
+def prepare_empty_pow_block(spec, rng=Random(3131)):
+    return spec.PowBlock(
```
Comment (zilm13) [inline tests/core/pyspec/eth2spec/test/helpers/pow_block.py:6]: Maybe `prepare_random_pow_block` is more correct?

**Review by zilm13 — COMMENTED** [review]
Looks good

File: tests/core/pyspec/eth2spec/test/merge/unittests/test_validate_merge_block.py [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+    else:
+        assert assertion_error_caught
+
+
+@with_merge_and_later
+@spec_state_test
+def test_validate_merge_block_success(spec, state):
+    pow_chain = prepare_random_pow_chain(spec, 2)
+    pow_chain[0].total_difficulty = spec.config.TERMINAL_TOTAL_DIFFICULTY - uint256(1)
+    pow_chain[1].total_difficulty = spec.config.TERMINAL_TOTAL_DIFFICULTY
```
Comment (zilm13) [inline tests/core/pyspec/eth2spec/test/merge/unittests/test_validate_merge_block.py:57]: For this and following this kind of test setup could look more readable I think
```suggestion
    pow_chain[-1].total_difficulty = spec.config.TERMINAL_TOTAL_DIFFICULTY
    pow_chain[-2].total_difficulty = spec.config.TERMINAL_TOTAL_DIFFICULTY - uint256(1)
```

Comment (mkalinin) [reply tests/core/pyspec/eth2spec/test/merge/unittests/test_validate_merge_block.py:57]: An improve on that could be:
```python
pow_chain.head()
pow_chain.head(-1)
```

**Review by djrtwo — APPROVED** [review]
Looks good!

I'll let you decide how to address @zilm13's comments and then click "merge"

**Comment by mkalinin** [comment]
> Looks good!
> 
> I'll let you decide how to address @zilm13's comments and then click "merge"

Addressed. `PowChain` is my first class written in python

**Comment by djrtwo** [comment]
looks good!

**Pull request merged**